packages/amplify-graphql-schema-generator/src/ts-schema-generator/helpers.ts

```diff
@@ -147,19 +188,92 @@ export const createSchema = (schema: Schema, config?: DataSourceGenerateConfig):
     throw new Error('No valid tables found. Make sure at least one table has a primary key.');
   }
 
+  const engine = schema.getEngine().type;
+
+  const nullEnums = schema  // find null enums in all models
+    .getModels()
+    .map((model) => model
+        .getFields()
+        .filter((field) => field.type.kind === 'Enum')
+        .map((field) => {
+          if(field.type.kind === 'Enum'){
+            return createEnums(field.type, engine, model.getName(), field.name);
+          }
+          else{
+            return undefined;
+          }
+     }));
+
+    const notNullEnums = schema  // find Non null enums in all models
```

Comment:
naming nit: I think you mean "required" enums or "non null enum types"?

Reply:
I will update the naming to have more clear variable names to avoid confusion.